lib/Parse/ParseDecl.cpp

```diff
@@ -2798,46 +2770,22 @@ ParserResult<TypeDecl> Parser::parseDeclTypeAlias(bool WantDefinition,
     }
   }
 
-  // Parse optional inheritance clause.
-  // FIXME: Allow class requirements here.
-  SmallVector<TypeLoc, 2> Inherited;
-  if (isAssociatedType && Tok.is(tok::colon))
-    Status |= parseInheritance(Inherited, /*classRequirementLoc=*/nullptr);
-
   ParserResult<TypeRepr> UnderlyingTy;
-  if (WantDefinition || Tok.is(tok::equal)) {
-    if (Tok.is(tok::colon)) {
-      // It is a common mistake to write "typealias A : Int" instead of = Int.
-      // Recognize this and produce a fixit.
-      diagnose(Tok, diag::expected_equal_in_typealias)
-        .fixItReplace(Tok.getLoc(), "=");
-      consumeToken(tok::colon);
-    
-    } else if (parseToken(tok::equal, diag::expected_equal_in_typealias)) {
-      Status.setIsParseError();
-      return Status;
-    }
-    UnderlyingTy = parseType(diag::expected_type_in_typealias);
-    Status |= UnderlyingTy;
-    if (UnderlyingTy.isNull())
-      return Status;
-  }
-
-  // If this is an associated type, build the AST for it.
-  if (isAssociatedType) {
-    assert(!genericParams && "Associated types don't allow generic params");
-    auto assocType = new (Context) AssociatedTypeDecl(
-                                     CurDeclContext,
-                                     TypeAliasLoc, Id, IdLoc,
-                                     UnderlyingTy.getPtrOrNull());
-    assocType->getAttrs() = Attributes;
-    if (!Inherited.empty())
-      assocType->setInherited(Context.AllocateCopy(Inherited));
-    addToScope(assocType);
-    return makeParserResult(Status, assocType);
+  if (Tok.is(tok::colon)) {
```

Comment:
I'd appreciate if we could improve QoI here for cases where Swift 2-minded developers write "typealias A: P" or "typealias A" when they mean to use "associatedtype". It shouldn't be hard to identify such cases when we're in a protocol context, suggest "associatedtype", and finish processing it as an associated type.